### PR TITLE
Added element-wise multiplication to Matrices.

### PR DESCRIPTION
**Pull request by glenwayguy**

### CODE REVIEW

**Comment by Ellyster** [comment]
IMHO only ```hadamar``` is not so great name, since it can be confused with the Hadamar matrix (http://en.wikipedia.org/wiki/Hadamard_matrix), that is the one usually shorted to only ```Hadamar``` and it should be (if someday implemented):
 ```hadamar_matrix = Matrix.hadamar(original_matrix)```

I guess it need to be caled ```hadamar_product```,as @marcandre said is its most common name.

@michaelimstepf: If we cannot use the standard ```.*```, ```./``` and ```.**``` notation in Ruby that would be the ideal, what about ```&*```, ```&/``` and ```&**``` as shorthands?

@hsbt: Is a really common operation, for example, right now I'm using it to select irregular sections of a matrix by doing the hadamard product of te matrix with one containing '1's in the areas that I want and '0's in the rest.

And is also very used in loosy compression, quantum information and communication algorithms.

File: lib/matrix.rb [hunk 1]
```diff
@@ -952,6 +952,33 @@ def /(other)
...
   #
+  # Hadamard product
+  #    Matrix[[1,2], [3,4]].hadamard(Matrix[[1,2], [3,2]])
+  #      => 1  4
+  #         9  8
+  #
+  def hadamard(m)
+    case m
+    when Numeric
+      Matrix.Raise ErrOperationNotDefined, "element_multiplication", self.class, m.class
```
Comment (Ellyster) [inline lib/matrix.rb]: Matrix.Raise ErrOperationNotDefined, "hadamard_product", self.class, m.class

**Comment by marcandre** [comment]
Thanks @hsbt for assigning this to me, I should have done that (and not forget about this!)

I'm travelling right now, will add `hadamard_product` and an alias `entrywise_product` in a few weeks at the most.

**Comment by glenwayguy** [comment]
@marcandre Hi, is this still good to be added, or do any other changes need to be made?

**Comment by marcandre** [comment]
Argh, completely slipped my mind to merge this when I got back, very sorry everyone, this won't make it in the next Ruby I'm afraid, we're long passed the feature freeze...

**Comment by luchillo17** [comment]
Was this never merged?

**Comment by marcandre** [comment]
Thanks @luchillo17  for the ping, and sorry for keeping on forgetting about this.

I some commits ready for this and https://bugs.ruby-lang.org/issues/10903 : in #1746

Any comments before I merge in trunk?

**Comment by marcandre** [comment]
Merged. Thanks again

**Pull request closed without merging**